src/mtcp/mtcp_restart.c

```diff
@@ -861,6 +862,20 @@ static int read_one_memory_area(int fd)
     }
   }
 
+#ifdef FAST_RST_VIA_MMAP
+    /* CASE MAP_ANONYMOUS with FAST_RST enabled
+     * We only want to do this in the MAP_ANONYMOUS case, since we don't want
+     *   any writes to RAM to be reflected back into the underlying file.
+     * Note that in order to map from a file (ckpt image), we must turn off
+     *   anonymous (~MAP_ANONYMOUS).  It's okay, since the fd
+     *   should have been opened with read permission, only.
+     */
+    else if (area.flags & MAP_ANONYMOUS) {
+      mmapfile (fd, area.addr, area.size, area.prot | PROT_WRITE,
```

Comment:
The reason why we add write permissions is still unclear to me. Do we remove these permissions later? If not, it seems like this conflicts with the open mode of the file which, as the comment above suggests, should have been read-only.